lib/web/join_tokens.go

```diff
@@ -239,21 +229,27 @@ func createJoinToken(ctx context.Context, m nodeAPIGetter, roles types.SystemRol
 	}, nil
 }
 
-func getJoinScript(settings scriptSettings, m nodeAPIGetter) (string, error) {
-	// Skip decoding validation for IAM tokens since they are generated with a different method
-	if settings.joinMethod != string(types.JoinMethodIAM) {
-		// This token does not need to be validated against the backend because it's not used to
-		// reveal any sensitive information. However, we still need to perform a simple input
-		// validation check by verifying that the token was auto-generated.
-		// Auto-generated tokens must be encoded and must have an expected length.
-		decodedToken, err := hex.DecodeString(settings.token)
-		if err != nil {
-			return "", trace.Wrap(err)
+func getJoinScript(ctx context.Context, settings scriptSettings, m nodeAPIGetter) (string, error) {
+	switch settings.joinMethod {
+	case string(types.JoinMethodUnspecified), string(types.JoinMethodToken), string(types.JoinMethodIAM):
+		if settings.joinMethod != string(types.JoinMethodIAM) {
+			decodedToken, err := hex.DecodeString(settings.token)
+			if err != nil {
+				return "", trace.Wrap(err)
+			}
+			if len(decodedToken) != auth.TokenLenBytes {
+				return "", trace.BadParameter("invalid token %q", decodedToken)
+			}
 		}
 
```

Comment:
If you pull the `GetToken` part out of the switch, you can simplify here, and can do less typecasting either way
 ```suggestion
	switch types.JoinMethod(settings.joinMethod) {
	case types.JoinMethodUnspecified, types.JoinMethodToken:
		decodedToken, err := hex.DecodeString(settings.token)
		if err != nil {
			return "", trace.Wrap(err)
		}
		if len(decodedToken) != auth.TokenLenBytes {
			return "", trace.BadParameter("invalid token %q", decodedToken)
		}
	case types.JoinMethodIAM:
```